src/materializer_vnode.erl

```diff
@@ -444,18 +488,19 @@ fetch_updates_from_cache(OpsCache, Key) ->
     end.
 
 -spec materialize_snapshot(txid() | ignore, key(), type(), snapshot_time(), boolean(), state(), snapshot_get_response())
-    -> {ok, snapshot_time()} | {error, reason()}.
+    -> {ok, {snapshot_time(), snapshot()}} | {error, reason()}.
 
-materialize_snapshot(_TxId, _Key, _Type, _SnapshotTime, _ShouldGC, _State, #snapshot_get_response{
+materialize_snapshot(_TxId, _Key, _Type, _MinSnapshotTime, _ShouldGC, _State, #snapshot_get_response{
             number_of_ops=0,
-            materialized_snapshot=Snapshot
+            materialized_snapshot=Snapshot,
+            snapshot_time=CommitTime
```

Comment:
I would suggest renaming `CommitTime` here, because a snapshot corresponds to a vector clock time, and this time might *not* correspond to a commit time of a transaction. Consider e.g. the case where a snapshot consists of changes resulting from two concurrent txns. What would be the commit time in this case if both transaction's commit vectors are incomparable?

Reply:
Indeed, I chose a name more close to the actual naming convention here and named it `SnapshotTime` hope it is correct ?

I prefer to not introduce the term UniqueToken here and keep this in the validate_or_read API.